src/tblite/calculator.F90

```diff
@@ -474,18 +474,18 @@ subroutine num_grad_chrg(env, mol, tblite)
    step2 = 0.5_wp / step
    call get_ceh(env,mol,tblite)
    
-   !$omp parallel do private(j,cehr,cehl) shared(env, numgrad, mol, tblite, step, step2) 
+   !$omp parallel do private(j, cehr, cehl, local_mol) shared(env, numgrad, mol, tblite, step, step2)
    do i = 1, mol%n
+      local_mol = mol    ! create a thread-private copy of mol
```

Comment:
Probably, you want to `call local_mol%copy(mol)`. At least it is done in this way in `src/type/calculator.f90`

Reply:
In general, this is not necessary, because the Molecule type only contains allocatable arrays and no pointer references. And the underlying data of arrays is copied when you do an assignment. However, the assignment operator is overwritten for the Molecule type. Not sure how this influences this.